src/tqec/sketchup/block_graph.py

```diff
@@ -0,0 +1,551 @@
+"""Graph representation of a 3D spacetime defect diagram by explicit blocks."""
+
+from __future__ import annotations
+
+from io import BytesIO
+import typing as ty
+import pathlib
+from dataclasses import dataclass, astuple
+from enum import Enum
+
+import networkx as nx
+
+from tqec.position import Position3D, Direction3D
+from tqec.sketchup.zx_graph import (
+    ZXGraph,
+    NodeType,
+    ZXNode,
+    ZXEdge,
+)
+from tqec.exceptions import TQECException
+
+
+@dataclass(frozen=True)
+class Color3D:
+    """Get face colors along the x, y, and z axes."""
+
+    x: str | None
+    y: str | None
+    z: str | None
```

Comment:
As a first improvement, I would change the typing to either
```suggestion
    x: typing.Literal["x", "z"] | None
    y: typing.Literal["x", "z"] | None
    z: typing.Literal["x", "z"] | None
```
or 
```suggestion
    x: typing.Literal["x", "z", "o"]
    y: typing.Literal["x", "z", "o"]
    z: typing.Literal["x", "z", "o"]
```
It might be worth representing one colour as a small enumeration such as
```py
class Colour(Enum):
    X = "x"
    Z = "z"
    O = "o"
```
and define some helper methods on that enumeration. I leave the choice to you :)